### PR TITLE
Finish the book split.

### LINKED ISSUE #24: [Tooling Suggestion] Split Handbook into multiple topical files
At the moment, the Handbook is a slightly unwieldy ~1600 lines and 92KB. I think there might be some benefit to splitting the Handbook up into several smaller topical documents.

My initial thought is to split the spell, equipment, and/or monster lists into their own document (or documents), much like the races and lore documents. I'm not 100% sold on the idea yet, but I figured I'd put it out there for perusal.

### LINKED ISSUE #40: Increase Text Font Size
So I decided to do a print test of M20 Space. It... didn't turn out well. I'm using the same font size as the M20 core, and it's not really a legible size on paper.

20180425_154308_hdr_edited

This is 8.5x11 paper in a 6.5MP image, by the way. I did a comparison with a D&D 4E rulebook, and the book's font is easily 1.5x the size. (It's also 300+ pages, though.)

The M20 font size is 95% of 8px (as much sense as that makes). Ideally, I'd want it at a minimum of 10px, if not more. My target is the paper being easily legible at arm's-length and rotated ~60 degrees away from the face.

I know you wanted to keep a maximum page count; this would need to increase that by about 10-20%.

I'll fiddle with the font size on M20 Space, and when I get something that works, I'll drop it here as another photo for consideration.

### PR DESCRIPTION
**Pull request by Web-eWorks**
This PR depends on #38, fixes #40, and closes #24 .

I moved the font size to a 9px base, unified the margins for text elements, (taking influence from Bootstrap, all text elements have `margin-top: 0; margin-bottom: 0.9em;`), rewrote the gulpfile to better pre-process the source files, wrote the handbook-combining task (and added a mechanism to control which files are included in the release), and generally updated things.

The introduction and SRD skills sections need their text re-flowed, but that's a job for you.

### CODE REVIEW

**Comment by Web-eWorks** [comment]
I got Travis to stop complaining, but the CodeClimate issues are up to you. I personally think the code complexity and line-length should be set to 20 and 30, respectively (a single-task function tends to be less easy to understand when all of it's guts are strewn across the file willy-nilly,) but that's just my opinion.

File: LICENSE [hunk 1]
```diff
@@ -1,3 +1,7 @@
+This product is 100% Open Game Content except for Product Identity, as per the
```
Comment (kgrubb) [inline LICENSE]: @Web-eWorks do you mind making a separate pull request for just the licensing, so its easier to document when/where the licensing was updated/added?

File: gulpfile.js [hunk 2]
```diff
@@ -2,32 +2,230 @@ const gulp = require('gulp');
...
+
+  const file = dataFiles[matches[0]];
+  if (file === undefined) {
+    console.log(`No such file ${matches[0]}.`);
+    return null;
+  }
+
+  // Get the table YAML document matching the key, or the first if no key.
+  let doc;
+  for (const d of file) if (d.id === matches[1]) doc = d;
```
Comment (kgrubb) [inline gulpfile.js]: Just adding the same comment from the last pull request (following our previous conversation), let's use array interators in this situation.

File: gulpfile.js [hunk 3]
```diff
@@ -2,32 +2,230 @@ const gulp = require('gulp');
...
+
+/*
+TODO: move this to separate files (make more extensible).
+-------------------------------------------------------------------------------
+*/
+
+dataTransform.push({
+  key: 'data',
+  // Parse the directive: $data <file> <key>
+  /* eslint no-restricted-syntax: "off", no-plusplus: "off" */
```
Comment (kgrubb) [inline gulpfile.js]: i'd rather us not override the eslint rules if possible.

File: gulpfile.js [hunk 4]
```diff
@@ -2,32 +2,230 @@ const gulp = require('gulp');
...
+  let body = '';
+
+  const columns = doc.columns || 1;
+  for (const h of doc.header) header += `| ${`${h} | `.repeat(columns - 1)}${h} |\n`;
+
+  const size = Math.floor(doc.data.length / columns);
+  const extra = doc.data.length % columns;
+
+  // Stack table entries side-by-side.
+  for (let i = 0; extra ? i <= size : i < size; i++) {
```
Comment (kgrubb) [inline gulpfile.js]: for-in is fine here, to resolve the `no-plusplus`, you can just put `i += 1` instead.

File: gulpfile.js [hunk 5]
```diff
@@ -2,32 +2,230 @@ const gulp = require('gulp');
...
+        .pipe(gulp.dest(config.out));
+    };
+    func.displayName = `compile-${d[0]}`;
+    tasks.push(func);
+  }
+  return gulp.parallel(tasks);
+}
+
+// eslint-disable-next-line arrow-body-style
+gulp.task('clean', () => {
```
Comment (kgrubb) [inline gulpfile.js]: to fix this eslint rule you can put something like this:

```js
gulp.task('clean', () => gulp.src('documents/*')
  .pipe(through.obj((file, err, callback) => {
    fs.unlinkSync(file.path);
    callback(null);
  })));
```

File: gulpfile.js [hunk 6]
```diff
@@ -2,32 +2,230 @@ const gulp = require('gulp');
...
+  for (const h of doc.header) header += `| ${`${h} | `.repeat(columns - 1)}${h} |\n`;
+
+  const size = Math.floor(doc.data.length / columns);
+  const extra = doc.data.length % columns;
+
+  // Stack table entries side-by-side.
+  for (let i = 0; extra ? i <= size : i < size; i++) {
+    let d = '| ';
+    for (let c = 0; c < columns && (i < size || c < extra); c++) {
+      const offset = c ? (c < extra ? c : extra) : 0;
```
Comment (kgrubb) [inline gulpfile.js]: nested ternaries can get super confusing really quickly, can we split this into two vars perhaps?

File: gulpfile.js [hunk 7]
```diff
@@ -47,8 +47,7 @@ function getDataFile(directive) {
   }
 
   // Get the table YAML document matching the key, or the first if no key.
-  let doc;
-  for (const d of file) if (d.id === matches[1]) doc = d;
+  const doc = file.find(d => d.id === matches[1]);
```
Comment (kgrubb) [inline gulpfile.js]: 😎 awesome!

**Comment by Web-eWorks** [comment]
I updated the gulpfile to use array functions where possible. I have eslint overrides in four places – one to allow the for loop needed to build the tables, and three that disable the arrow-function rule for better alignment of pipe statements. I'd appreciate a sign-off on those.

File: gulpfile.js [hunk 8]
```diff
@@ -2,32 +2,224 @@ const gulp = require('gulp');
...
+  const columns = doc.columns || 1;
+  doc.header.forEach((h) => {
+    header += `| ${`${h} | `.repeat(columns - 1)}${h} |\n`;
+  });
+
+  const size = Math.floor(doc.data.length / columns);
+  const extra = doc.data.length % columns;
+
+  // Stack table entries side-by-side.
+  /* eslint no-restricted-syntax: "off" */
```
Comment (kgrubb) [inline gulpfile.js]: I just pulled your latest changes, and it looks like this eslint override isn't necessary anymore :tada:. After removing this line and line 101 (swapping it back to error) locally, everything passes still.

File: gulpfile.js (hunk 3 above)
Comment (Web-eWorks) [reply gulpfile.js]: Can't get rid of the `no-restricted-syntax` one, I'm afraid. If you want to re-write the function to use array iterators, go right ahead; I've spent enough time on it already.

File: gulpfile.js [hunk 9]
```diff
@@ -2,32 +2,224 @@ const gulp = require('gulp');
...
+        .pipe(gulp.dest(config.out));
+    };
+    func.displayName = `compile-${d[0]}`;
+    tasks.push(func);
+  });
+  return gulp.parallel(tasks);
+}
+
+// eslint-disable-next-line arrow-body-style
+gulp.task('clean', () => gulp.src('documents/*')
```
Comment (kgrubb) [inline gulpfile.js]: this eslint override can be removed too 😎

**Comment by kgrubb** [comment]
😱 that last travis error is super weird. I think its just a fluke though.

**Comment by Web-eWorks** [comment]
I'll have to check, but I don't _think_ I'm doing anything weird. I'm gonna assume the phantomjs instance is consuming more memory than the VM has allocated, and it's crashing because of that.

Though, that's it's own problem.

**Comment by kgrubb** [comment]
hm, i'm able to replicate it locally:
screen shot 2018-05-08 at 12 38 40 am
Could it be something going on in `load-data`? I'll try to hunt down what's causing the OOM err locally in a sec.

**Comment by Web-eWorks** [comment]
Oh, I get it. I broke something (probably in the table compilation) and it's inf-looping.

**Comment by Web-eWorks** [comment]
Fixed.

**Comment by kgrubb** [comment]
:rocket: Awesome, this lgtm!

**Comment by kgrubb** [comment]
dang, i forgot about the squash merge option. Not a big deal though. Thanks for powering through the pr! Awesome work!

**Comment by Web-eWorks** [comment]
You're more than welcome. I'm extremely happy to help – after all, I'm the consumer of the work I'm doing.
